src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/RowDefinitionCollectionConverter.cs

```diff
@@ -0,0 +1,121 @@
+        /// An ArgumentNullException is thrown if the example object is null.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// An ArgumentException is thrown if the object is not null and is not a valid type,
+        /// or if the destinationType isn't one of the valid destination types.
+        /// </exception>
+        /// <param name="typeDescriptorContext"> The ITypeDescriptorContext for this call. </param>
+        /// <param name="cultureInfo"> The CultureInfo which is respected when converting. </param>
+        /// <param name="value"> The Thickness to convert. </param>
+        public override object ConvertFrom(ITypeDescriptorContext typeDescriptorContext, CultureInfo cultureInfo, object value)
+        {
+            if (value != null && value is string input)
+            {
+                IProvideValueTarget ipvt = typeDescriptorContext?.GetService(typeof(IProvideValueTarget)) as IProvideValueTarget;
+                Grid grid = ipvt?.TargetObject as Grid;
+                if (grid != null)
+                {
+                    RowDefinitionCollection collection = new RowDefinitionCollection(grid); // Pass Grid instance
+
+                    TokenizerHelper th = new TokenizerHelper(input, cultureInfo);
+                    while (th.NextToken())
+                    {
+                        collection.Add(new RowDefinition { Height = GridLengthConverter.FromString(th.GetCurrentToken(), cultureInfo) });
+                    }
+
+                    return collection;
+                }
+            }
+            throw GetConvertFromException(value);
+        }
+
+        /// <summary>
+        /// ConvertTo - Attempt to convert a RowDefinitionCollection to the given type
+        /// </summary>
+        /// <returns>
+        /// The object which was constructed.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// An ArgumentNullException is thrown if the example object is null.
+        /// </exception>
+        /// <param name="context"> The ITypeDescriptorContext for this call. </param>
+        /// <param name="culture"> The CultureInfo which is respected when converting. </param>
+        /// <param name="value"> The RowDefinitionCollection to convert. </param>
+        /// <param name="destinationType">The type to which to convert the RowDefinitionCollection instance. </param>
+        public override object ConvertTo(ITypeDescriptorContext typeDescriptorContext, CultureInfo cultureInfo, object value, Type destinationType)
+        {
+            ArgumentNullException.ThrowIfNull(value);
+            ArgumentNullException.ThrowIfNull(destinationType);
+            if (destinationType == typeof(string) && value is RowDefinitionCollection rowDefinitions)
+            {
+                char listSeparator = TokenizerHelper.GetNumericListSeparator(cultureInfo);
+                StringBuilder sb = new StringBuilder();
```

Comment:
The `StringBuilder` can have a defaulted capacity to the most common scenario, which imho is gonna be about `5 * rowDefinitions.Count`. This will prevent several re-allocations of the buffer. Same in the other converter.